src/biclustering_impl.h

```diff
@@ -30,45 +30,47 @@ class ConvexBiClustering {
     num_col_edges(D_col_.cols()),
     sp(show_progress_, D_row_.rows() + D_col_.cols()){
 
-    // Set initial values for optimization variables
-    U = X;
-    V_row = D_row * U;
-    Z_row = V_row;
-    V_col = D_col.transpose() * X.transpose();
-    Z_col = V_col;
-
-    P = Eigen::MatrixXd::Zero(n, p);
-    Q = Eigen::MatrixXd::Zero(n, p);
-
-    v_row_zeros = Eigen::ArrayXi::Zero(num_row_edges);
-    v_col_zeros = Eigen::ArrayXi::Zero(num_col_edges);
-    gamma = 0;
-
-    sp.set_v_norm_init(V_row.squaredNorm() + V_col.squaredNorm());
-
-    // Initialize storage buffers
-    buffer_size = 1.5 * (n + p);
-    UPath.resize(n * p, buffer_size);
-    V_rowPath.resize(p * num_row_edges, buffer_size);
-    V_colPath.resize(n * num_col_edges, buffer_size);
-    gamma_path.resize(buffer_size);
-    v_row_zeros_path.resize(num_row_edges, buffer_size);
-    v_col_zeros_path.resize(num_col_edges, buffer_size);
-
-    // Store initial values
-    nzeros_row = 0;
-    nzeros_col = 0;
-    storage_index = 0;
-    store_values();
-
-    // PreCompute chol(I_n + rho D_row^T D_row) and chol(I_p + rho D_col D_col^T) for easy inversions in the ADMM primal update steps
-    Eigen::MatrixXd IDTD_row = rho * D_row.transpose() * D_row + Eigen::MatrixXd::Identity(n, n);
-    row_primal_solver.compute(IDTD_row);
-
-    Eigen::MatrixXd IDDT_col = rho * D_col * D_col.transpose() + Eigen::MatrixXd::Identity(p, p);
-    col_primal_solver.compute(IDDT_col);
+      // Set initial values for optimization variables
+      U = X;
+      V_row = D_row * U;
+      Z_row = V_row;
+      V_col = U * D_col;
+      Z_col = V_col;
+
+      v_row_zeros = Eigen::ArrayXi::Zero(num_row_edges);
+      v_col_zeros = Eigen::ArrayXi::Zero(num_col_edges);
+      gamma = 0;
+
+      //precompute reused intermediates:
+      DDT_col = D_col * D_col.transpose();
+      DTD_row = D_row.transpose() * D_row;
+
+      //compute alpha
+      //TODO: implement a tighter alpha calculation
+      double row_max_deg = D_row.cwiseAbs().colwise().sum().maxCoeff();
+      double col_max_deg = D_col.cwiseAbs().rowwise().sum().maxCoeff();
+      alpha = 2 * (row_max_deg  * col_max_deg);
+
+
+      sp.set_v_norm_init(V_row.squaredNorm() + V_col.squaredNorm());
+
+      // Initialize storage buffers
+      buffer_size = 1.5 * (n + p);
+      UPath.resize(n * p, buffer_size);
+      V_rowPath.resize(p * num_row_edges, buffer_size);
+      V_colPath.resize(n * num_col_edges, buffer_size);
```

Comment:
Dumb Q: Does this need to be transposed since we're plumbed a transpose in `V_col` through this PR?

Reply:
I don't know the semantics of this well enough to be able to tell you

Reply:
Hmmmm.... I have to plumb through it, but the core is here: 

https://github.com/DataSlingers/clustRviz/blob/0f0acbe19d712671a602d674afd14ff3414d531c/src/biclustering_impl.h#L310

Essentially, we take the `V_row` matrix and read it as a vector. (The `Eigen::Map` stuff is setting up a read-only proxy from already allocated memory in `V_row`.)  We take that vector and put it as a row in the data that gets pushed back into `R`. I'm concerned that with the transposes and switching data layout, we are going to get the edges messed back up on the `R` end of things.

Testing that we get essentially the same answer before and after pr with `exact = TRUE` should suffice here.